From 1387444e1beab371721efc4259a4c2f24f9a9348 Mon Sep 17 00:00:00 2001
From: Hiroshi Miura <miurahr@nttdata.co.jp>
Date: Wed, 27 Jul 2016 15:20:10 +0900
Subject: [PATCH] Don't run project team structure when --no-team CLI option is
 specified.

                   - option no-team obviously means working project without team, so asking users a conversion to
                     4.0 style team is bad behavior.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>
---
 src/org/omegat/convert/ConvertProject26to37team.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/org/omegat/convert/ConvertProject26to37team.java b/src/org/omegat/convert/ConvertProject26to37team.java
index ae56bfb..c881801 100644
--- a/src/org/omegat/convert/ConvertProject26to37team.java
+++ b/src/org/omegat/convert/ConvertProject26to37team.java
@@ -37,6 +37,7 @@ import org.eclipse.jgit.lib.Repository;
 import org.eclipse.jgit.lib.StoredConfig;
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.revwalk.RevWalk;
+import org.omegat.CLIParameters;
 import org.omegat.core.Core;
 import org.omegat.core.data.ProjectProperties;
 import org.omegat.core.team2.ProjectTeamSettings;
@@ -66,6 +67,12 @@ public class ConvertProject26to37team {
     public static void checkTeam(File projectRootFolder) throws Exception {
         if (isSVNDirectory(projectRootFolder) || isGITDirectory(projectRootFolder)) {
             // project is 2.6-style team project
+
+            // When --no-team option is given, we skip conversion silently.
+            if (Core.getParams().containsKey(CLIParameters.NO_TEAM)) {
+                return;
+            }
+
             if (isConsoleMode()) {
                 Core.getMainWindow().displayWarningRB("TEAM_26_TO_37_CONSOLE");
                 return;
-- 
2.9.3

